src/movies.js

```diff
@@ -1,16 +1,194 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(moviesData){
+  let allDirectors = [];
+  allDirectors = moviesData
+  .map(function (movie){
+    
+    return movie.director;
+  })
+  .filter(function(d, i, self){
+    return i == self.indexOf(d);
+  });
+return allDirectors;
```

Comment:
Nice! I think you could simplify this slightly without the second part. Also it's good practice to name the variables as opposed to using letters. I don't know what each of these values is doing!